docs/source/CartesianConfig.rst

```diff
@@ -363,6 +363,91 @@ statements addressing all objects. Note: This is contrary to the way the
 Cartesian configuration file as a whole is parsed (top-down).
 
 
+.. _heterogeneous_guest_variants:
+
+Heterogeneous guest variants
+============================
+
+The ‘``join``’ filter combined with the ‘``suffix``’ operator can be utilized
+together in order to produce guest variants with different guest OS or other
+types of configuration within the same test. Adding the ‘``suffix``’ keyword
+at the end of each variant to be combined and joining all the variants in the
+end will produce one final variant product with all participating variant
+dictionaries separately suffixed and combined into one final dictionary. For
+instance the following definition
+
+::
+
+    variants:
+        - one:
+            key1 = Hello
+            suffix _v1
+        - two:
+            key1 = Bye
+            key2 = World
+            suffix _v2
+        - three:
+    variants:
+        - four:
+            key3 = foo
+            only one
+        - five:
+            key3 = bar
+        - six:
+            key1 = foo
+            key2 = bar
+            only two
+
+    join one two
+
+will join variants one and two with suffixed parameters ``key1_v1 = Hello``
+and ``key1_v2 = Bye`` in each of the following resulting dictionaries:
+
+::
+
+    dict    1:  four.one.five.two
+        dep = []
+        key1 = Hello
+        key2 = World
+        key3 = bar
+        name = four.one.five.two
+        shortname = four.one.five.two
+    dict    2:  four.one.six.two
+        dep = []
+        key1 = foo
+        key1_v1 = Hello
+        key2 = bar
+        key2_v2 = World
+        key3 = foo
+        name = four.one.six.two
+        shortname = four.one.six.two
+    dict    3:  five.one.two
+        dep = []
+        key1 = Hello
+        key2 = World
+        key3 = bar
+        name = five.one.two
+        shortname = five.one.two
+    dict    4:  five.one.six.two
+        dep = []
+        key1 = foo
+        key1_v1 = Hello
+        key2 = bar
+        key2_v2 = World
+        key3 = bar
+        name = five.one.six.two
+        shortname = five.one.six.two
+
```

Comment:
Ï have got different resulting dictionaries from yours that make more sense to me (according to your input input):

```
dict    1:  four.one.five.two
    dep = []
    key1_v1 = Hello
    key1_v2 = Bye
    key2 = World
    key3 = bar
    name = four.one.five.two
    shortname = four.one.five.two
dict    2:  four.one.six.two
    dep = []
    key1 = foo
    key1_v1 = Hello
    key1_v2 = Bye
    key2 = bar
    key2_v2 = World
    key3 = foo
    name = four.one.six.two
    shortname = four.one.six.two
dict    3:  five.one.two
    dep = []
    key1_v1 = Hello
    key1_v2 = Bye
    key2 = World
    key3 = bar
    name = five.one.two
    shortname = five.one.two
dict    4:  five.one.six.two
    dep = []
    key1 = foo
    key1_v1 = Hello
    key1_v2 = Bye
    key2 = bar
    key2_v2 = World
    key3 = bar
    name = five.one.six.two
    shortname = five.one.six.two
```

see e.g. suffixes in ``dict 1``

Reply:
I also get this output...

Reply:
I am not sure what went wrong when I was pasting the output but indeed, this is the one I originally intended. Sorry for the clipboard mismanagement! All fixed now.